torch/nn/attention/_flex_attention.py

```diff
@@ -93,11 +151,22 @@ def score_mod(
 
     """
 
+    if block_sparse_mask is None:
+        block_sparse_mask = _create_empty_block_sparse_mask(query.device)
     if torch.compiler.is_dynamo_compiling():
         # mark head_dim always to be static
         for x in [query, key, value]:
             torch._dynamo.mark_static(x, -1)
-        out, _ = flex_attention_hop(query, key, value, score_mod)
+        out, _ = flex_attention_hop(
+            query,
+            key,
+            value,
```

Comment:
do we need to flatten the mask at this level?

Reply:
I think HOO doesn't support arbitrary argument types, so we have to flatten?